build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/PruneChangelogsTask.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.release;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.elasticsearch.gradle.VersionProperties;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.process.ExecOperations;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+
+/**
+ * Once a minor release has happened, we no longer need to keep the changelog files that went into
+ * that release in the development branch for that major series or the branch for the next major
+ * series
+ * <p>
+ * This last examines the git history in order to work out which files can be deleted, and
+ * does the deletion after confirming with the user.
+ */
+public class PruneChangelogsTask extends DefaultTask {
+    private static final Logger LOGGER = Logging.getLogger(PruneChangelogsTask.class);
+
+    private final ConfigurableFileCollection changelogs;
+    private final GitWrapper gitWrapper;
+    private final Path rootDir;
+
+    private boolean confirmed;
+
+    @Inject
+    public PruneChangelogsTask(Project project, ObjectFactory objectFactory, ExecOperations execOperations) {
+        changelogs = objectFactory.fileCollection();
+        gitWrapper = new GitWrapper(execOperations);
+        rootDir = project.getRootDir().toPath();
+    }
+
+    @InputFiles
```

Comment:
I'm not sure it's actually worth annotating these things with `@Input` since we don't ever actually want this task to be UP-TO-DATE given that part of the "input" is git information which we only have at execution time. I'd just replace these all with `@Internal` and save the overhead of having Gradle snapshot these inputs.